thrust/thrust/iterator/detail/universal_categories.h

```diff
@@ -115,4 +114,5 @@ struct random_access_universal_iterator_tag
   }
 };
 
+_CCCL_SUPPRESS_DEPRECATED_POP
 THRUST_NAMESPACE_END
```

Comment:
This should be after the end of the namespace

Reply:
I talked to @dkolsen-pgi about how pragmas work and they are always attached to the entity following the pragma. So by putting `_CCCL_SUPPRESS_DEPRECATED_POP` before the namespace end, I believe the pragma pop is attached to the closing of the namespace. If I would put it afterwards, then the pragma pop is attached to whatever entity happens to end up afterwards from any other header or source file.

The compiler later reconstructs the range between pragma push and pop by searching the AST, and this is sometimes buggy. I observed less bugs by putting the pragma pop before the namespace end. This is why I do it this way.